--- xmlada-gpl-2015-src/xmlada-config.in	2014-09-29 10:38:49.000000000 -0700
+++ new/xmlada-config.in	2015-09-03 14:29:48.876634216 -0700
@@ -6,35 +6,27 @@
 prefix=`dirname $prefix`
 
 exec_prefix=@exec_prefix@
-libdir=@libdir@/xmlada
+libdir=@libdir@
 
-cflags_static=""
-cflags_relocatable=""
-mflags_static=""
-mflags_relocatable=""
-libs_relocatable=""
+cflags_static="-I@includedir@"
+cflags_relocatable="-I@includedir@"
+mflags_static="-aI@includedir@ -aO${libdir}/static"
+mflags_relocatable="-aI@includedir@ -aO${libdir}/xmlada/relocatable"
+libs_relocatable="-L${libdir}/relocatable"
 libs_static=""
 full_libs_relocatable=""
 full_libs_static=""
 
-for module in dom sax schema unicode input; do
-    cflags_static="${cflags_static} -I@includedir@/xmlada_${module}.static"
-    cflags_relocatble="${cflags_relocatable} -I@includedir@/xmlada_${module}.relocatable"
-
-    mflags_static="${mflags_static} -aI@includedir@/xmlada_${module}.static -aO@libdir@/xmlada_${module}.static"
-    mflags_relocatable="${mflags_relocatable} -aI@includedir@/xmlada_${module}.relocatable -aO@libdir@/xmlada_${module}.relocatable"
-done
-
 for module in sax unicode; do
-   libs_static="${libs_static} @libdir@/xmlada_${module}.static/libxmlada_${module}.a"
-   libs_relocatable="${libs_relocatable} -L@libdir@/xmlada_${module}.relocatable -lxmlada_${module}"
+   libs_static="${libs_static} ${libdir}/static/libxmlada_${module}.a"
+   libs_relocatable="${libs_relocatable} -lxmlada_${module}"
 done
 
-libs_static="${libs_static} @libdir@/xmlada_input.static/libxmlada_input_sources.a"
-libs_relocatable="${libs_relocatable} -L@libdir@/xmlada_input.relocatable -lxmlada_input_sources"
+libs_static="${libs_static} ${libdir}/static/libxmlada_input_sources.a"
+libs_relocatable="${libs_relocatable} -lxmlada_input_sources"
 
 for module in dom schema; do
-    full_libs_static="${full_libs_static} @libdir@/xmlada_${module}.static/libxmlada_${module}.a"
+    full_libs_static="${full_libs_static} ${libdir}/static/libxmlada_${module}.a"
     full_libs_relocatable="${full_libs_relocatable} -lxmlada_${module}"
 done
 
