--- gprbuild-2011-src.orig/gnat/gnatvsn.ads	2011-04-28 07:21:24.000000000 -0700
+++ gprbuild-2011-src/gnat/gnatvsn.ads	2012-04-13 23:10:19.129813192 -0700
@@ -51,7 +51,7 @@
    type Gnat_Build_Type is (Gnatpro, FSF, GPL);
    --  See Get_Gnat_Build_Type below for the meaning of these values
 
-   Build_Type : constant Gnat_Build_Type := GPL;
+   Build_Type : constant Gnat_Build_Type := FSF;
    --  Kind of GNAT Build:
    --
    --    FSF
@@ -81,7 +81,7 @@
    --  Return the name of the Copyright holder to be displayed by the different
    --  GNAT tools when switch --version is used.
 
-   Ver_Len_Max : constant := 64;
+   Ver_Len_Max : constant := 256;
    --  Longest possible length for Gnat_Version_String in this or any
    --  other version of GNAT. This is used by the binder to establish
    --  space to store any possible version string value for checks. This
