--- gprbuild-gpl-2014-src/src/gprinstall-install.adb	2014-04-17 02:50:17.000000000 -0700
+++ gprbuild-gpl-2014-src-new/src/gprinstall-install.adb	2014-12-12 11:46:16.072999664 -0800
@@ -30,6 +30,7 @@
 with Ada.Strings.Fixed;                      use Ada.Strings;
 with Ada.Strings.Unbounded;                  use Ada.Strings.Unbounded;
 with Ada.Text_IO;                            use Ada.Text_IO;
+with Gprbuild_OS_Lib;                        use Gprbuild_OS_Lib;
 
 with GNAT.MD5;    use GNAT.MD5;
 with Gpr_Util;    use Gpr_Util;
--- gprbuild-gpl-2014-src/src/gprslave.adb	2014-04-17 02:50:17.000000000 -0700
+++ gprbuild-gpl-2014-src-new/src/gprslave.adb	2014-12-12 14:43:31.308822806 -0800
@@ -34,6 +34,7 @@
 with Ada.Text_IO;                           use Ada.Text_IO;
 with Interfaces;
 with System.Multiprocessors;                use System;
+with Gprbuild_OS_Lib;                        use Gprbuild_OS_Lib;
 
 with Csets;                         use Csets;
 with Gnatvsn;                       use Gnatvsn;
--- gprbuild-gpl-2014-src/gprbuild.gpr	2014-12-12 14:54:47.944875176 -0800
+++ gprbuild-gpl-2014-src-new/gprbuild.gpr	2014-12-12 14:54:35.484874211 -0800
@@ -102,7 +102,8 @@
 
    package Linker is
       Common_Switches := (project'Object_Dir & "/gprbuild_dummies.o",
-                          project'Object_Dir & "/link.o");
+                          project'Object_Dir & "/link.o",
+		          project'Object_Dir & "/gprbuild_os_lib_c.o");
       --  We use project'Object_Dir here instead of GprBuild'Object_Dir
       --  for compatibility with GNAT Pro 6.0.2.
 
