--- a/asis/a4g-a_opt.ads
+++ b/asis/a4g-a_opt.ads
@@ -59,7 +59,7 @@
    --  treated as an error: ASIS_Failed is raised and the warning message is
    --  sent to an ASIS Diagnosis string.
 
-   Strong_Version_Check : Boolean := True;
+   Strong_Version_Check : Boolean := False;
    --  Strong version check means that version strings read from the tree and
    --  stored in Gnatvsn are compared. Weak check means comparing ASIS version
    --  numbers. See BA23-002
Index: b/tools/tool_utils/asis_ul-source_table-processing.adb
===================================================================
--- a/tools/tool_utils/asis_ul-source_table-processing.adb
+++ b/tools/tool_utils/asis_ul-source_table-processing.adb
@@ -377,7 +377,7 @@
       Add_Needed_Sources : Boolean := Mimic_gcc)
    is
    begin
-      Asis.Implementation.Initialize ("-k -ws -asis05 -sv");
+      Asis.Implementation.Initialize ("-k -ws -asis05");
 
       if True then -- Can we get rid of Only_Bodies????
          Process_Sources_From_Table
