--- adactl-1.19r10/Makefile	2017-12-14 11:35:01.000000000 -0800
+++ adactl-1.19r10-new/Makefile	2018-02-01 10:34:08.598013854 -0800
@@ -24,6 +24,7 @@
 	@echo "--              | adactl     -- build adactl                 --"
 	@echo "--              | ptree      -- build ptree                  --"
 	@echo "--              | pfni       -- build pfni                   --"
+	@echo "--              | adactl_fix -- build adactl_fix             --"
 	@echo "--              | install    -- install everything           --"
 	@echo "--              | clean      -- delete object files          --"
 	@echo "--              | veryclean  -- clean + delete executable    --"
@@ -31,7 +32,7 @@
 	@echo "---------------------------------------------------------------"
 
 
-build : adactl ptree pfni;
+build : adactl ptree pfni adactl_fix
 
 adactl :
 	gprbuild build.gpr adactl ${GARGS} -cargs ${CARGS} -bargs ${BARGS} -largs ${LARGS}
@@ -42,8 +43,11 @@
 pfni :
 	gprbuild build.gpr pfni   ${GARGS} -cargs ${CARGS} -bargs ${BARGS} -largs ${LARGS}
 
+adactl_fix:
+	gprbuild build.gpr adactl_fix ${GARGS} -cargs ${CARGS} -bargs ${BARGS} -largs ${LARGS}
+
 install :
-	gprinstall build.gpr -f
+	gprinstall --prefix=${PREFIX} -p build.gpr -f
 
 clean :
 	gprclean -c build.gpr
