diff -ru ahven-2.4/Makefile ahven-2.4-new/Makefile
--- ahven-2.4/Makefile	2014-02-08 13:21:51.000000000 -0800
+++ ahven-2.4-new/Makefile	2014-12-18 10:28:36.088761898 -0800
@@ -93,11 +93,11 @@
 install_lib: build_lib
 	mkdir -p $(PREFIX)/include/ahven
 	mkdir -p $(LIBDIR)/ahven
-	mkdir -p $(PREFIX)/lib/gnat
+	mkdir -p $(LIBDIR)/gnat
 	$(INSTALL) -m 644 $(SOURCES) $(PREFIX)/include/ahven
 	$(INSTALL) -m 444 $(ALI_FILES) $(LIBDIR)/ahven
 	$(INSTALL) -m 644 lib/$(STATIC_LIBRARY) $(LIBDIR)/ahven
-	$(INSTALL) -m 644 $(GPR_FILE) $(PREFIX)/lib/gnat
+	$(INSTALL) -m 644 $(GPR_FILE) $(LIBDIR)/gnat
 
 install_docs: userguide
 	mkdir -p $(PREFIX)/share/doc/ahven
