diff -ur gtkada-git12cb7d7f33e.org/configure gtkada-git12cb7d7f33e/configure
--- gtkada-git12cb7d7f33e.org/configure	2012-04-04 01:06:54.000000000 -0700
+++ gtkada-git12cb7d7f33e/configure	2012-04-08 12:14:24.792477066 -0700
@@ -3508,7 +3508,7 @@
 # Ignore user's choice of libdir, as well as configure's own default,
 # because we want GtkAda to always install in /lib (backward compatibility)
 
-libdir='${prefix}/lib'
+#libdir='${prefix}/lib'
 
                                                                                                                                                                 ac_config_files="$ac_config_files Makefile src/Makefile.common src/Makefile src/tools/gtkada-config src/tools/Makefile testgtk/Makefile src/gtkextra/Makefile src/opengl/Makefile src/gnome/Makefile docs/Makefile docs/gtkada_rm/Makefile po/Makefile src/config.h projects/gtkada.pc projects/gtkada_install_dynamic.gpr projects/gtkada_install_dynamic_gl.gpr"
 cat >confcache <<\_ACEOF
diff -ur gtkada-git12cb7d7f33e.org/projects/Makefile gtkada-git12cb7d7f33e/projects/Makefile
--- gtkada-git12cb7d7f33e.org/projects/Makefile	2012-04-04 01:06:54.000000000 -0700
+++ gtkada-git12cb7d7f33e/projects/Makefile	2012-04-08 16:53:59.853051061 -0700
@@ -4,7 +4,7 @@
 
 SOVER=${MAJOR}.${MINOR}
 
-prjdir="${DESTDIR}${prefix}/lib/gnat"
+prjdir="${DESTDIR}${libdir}/gnat"
 
 install: force
 	${MKDIR} ${prjdir}
Only in gtkada-git12cb7d7f33e/src: .Makefile.in.swp
