--- gtkada-2.24.3g40fc04b/configure	2012-10-30 07:55:59.000000000 -0700
+++ gtkada-2.24.3g40fc04b.new/configure	2012-11-15 22:02:54.275187830 -0800
@@ -3959,7 +3959,7 @@
 # Ignore user's choice of libdir, as well as configure's own default,
 # because we want GtkAda to always install in /lib (backward compatibility)
 
-libdir='${prefix}/lib'
+#libdir='${prefix}/lib'
 
 ac_config_files="$ac_config_files Makefile src/Makefile.common src/Makefile src/tools/gtkada-config src/tools/Makefile testgtk/Makefile src/gtkextra/Makefile src/opengl/Makefile src/gnome/Makefile docs/Makefile docs/gtkada_rm/Makefile po/Makefile src/config.h projects/gtkada.pc projects/gtkada_install_dynamic.gpr projects/gtkada_install_dynamic_gl.gpr"
 
--- gtkada-2.24.3g40fc04b/projects/Makefile	2012-10-30 07:55:59.000000000 -0700
+++ gtkada-2.24.3g40fc04b.new/projects/Makefile	2012-11-15 21:57:26.276199550 -0800
@@ -8,7 +8,7 @@
 # from the command line, so we cannot include DESTDIR directly here
 # otherwise they have to call make with
 #     make DESTDIR=/foo PRJDIR=/foo/usr/share/gpr
-PRJDIR=${prefix}/lib/gnat
+PRJDIR=${libdir}/gnat
 
 install: force
 	${MKDIR} "${DESTDIR}${PRJDIR}"
--- gtkada-2.24.3g40fc04b/testgtk/Makefile.in	2012-10-30 07:55:59.000000000 -0700
+++ gtkada-2.24.3g40fc04b.new/testgtk/Makefile.in	2012-11-15 22:17:41.865156080 -0800
@@ -32,7 +32,7 @@
 endif
 
 GNATMAKELDFLAGS = -largs -L$(src_dir)/lib-obj -L$(opengl_dir) -lgtkada \
-  $(GTKADA_GL_LIBS) @GTK_LIBS_FOR_GNATMAKE@ $(TARGET_LFLAGS)
+  $(GTKADA_GL_LIBS) @GTK_LIBS_FOR_GNATMAKE@ -lm $(TARGET_LFLAGS)
 
 all: opengl/view_gl.adb $(OPENGL_C_OBJECTS) $(target) 
 
