--- gtkada-gpl-2015-src/testgtk/testgtk.gpr	2015-06-01 07:42:24.000000000 -0700
+++ n/testgtk/testgtk.gpr	2015-09-04 15:58:57.368570179 -0700
@@ -24,6 +24,9 @@
 
    end Builder;
 
+   package Linker is
+       for Required_Switches use ("-lX11", "-lm");
+   end Linker;
    package Compiler is
       --  subprogram specs not required in testgtk
       for Switches ("Ada") use Shared.Compiler'Switches ("Ada") & ("-gnaty-s");
