--- gnatcoll-db-22.0.0/pgxs/source/pgxs-types.ads	2021-11-13 15:13:44.062816015 -0800
+++ gnatcoll-db/pgxs/source/pgxs-types.ads	2021-11-13 14:55:53.542754251 -0800
@@ -34,6 +34,7 @@
 --  varchar                    VarChar*  postgres.h  Var_Char
 
 with Interfaces.C;
+with Interfaces.C.Extensions;
 
 with PGXS.Varlen;
 
@@ -41,7 +42,7 @@
 
    type Oid is private;
 
-   subtype Bool is Interfaces.C.C_bool;
+   subtype Bool is Interfaces.C.Extensions.bool;
 
    subtype Int_16 is Interfaces.Integer_16;
 
--- gnatcoll-db-22.0.0/pgxs/source/pgxs-composites.ads	2021-09-21 06:09:40.000000000 -0700
+++ gnatcoll-db/pgxs/source/pgxs-composites.ads	2021-11-13 15:29:08.142869331 -0800
@@ -91,7 +91,7 @@
    type Attributes_Arrays (Size : Attribute_Count) is record
       Descriptor : PGXS.Tuple_Desc;
       Datums     : Datum_Array (1 .. Size);
-      Nulls      : Bool_Array (1 .. Size) := (others => Interfaces.C.True);
+      Nulls      : Bool_Array (1 .. Size) := (others => Interfaces.C.Extensions.bool(True));
    end record;
 
    type Attributes is access all Attributes_Arrays
--- gnatcoll-db-23.0w/pgxs/source/pgxs-composites.adb	2022-01-17 06:31:55.000000000 -0800
+++ nbd/pgxs/source/pgxs-composites.adb	2022-05-28 18:10:04.772830822 -0700
@@ -75,7 +75,7 @@
      (Self  : in out Attributes;
       Index : Attribute_Number) is
    begin
-      Self.Nulls (Index) := Interfaces.C.True;
+      Self.Nulls (Index) := Interfaces.C.Extensions.bool(True);
    end Set_Null;
 
    ---------------
@@ -87,7 +87,7 @@
       Index : Attribute_Number;
       To    : PGXS.Datum) is
    begin
-      Self.Nulls (Index) := Interfaces.C.False;
+      Self.Nulls (Index) := Interfaces.C.Extensions.bool(False);
       Self.Datums (Index) := To;
    end Set_Value;
 
