diff -ru SparForte-2.1/configure SparForte-2.1-new/configure
--- SparForte-2.1/configure	2017-09-01 17:51:22.000000000 -0700
+++ SparForte-2.1-new/configure	2017-09-19 13:10:13.213717488 -0700
@@ -1202,25 +1202,6 @@
 sleep 1
 
 echo -n "apq library: checking for one or more database config scripts... "
-if [ -n "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
-   echo "no"
-else
-   if [ -z "$HAS_POSTGRES" ] ; then
-      if [ -z "$HAS_MYSQL" ] ; then
-         echo "no"
-         echo
-         echo "The APQ library uses the pg_config and mysql_config scripts to configure"
-         echo "itself.  Not all operating systems include these, or they include them"
-         echo "as part of the 'devel' packages.  If you don't have these scripts, you"
-         echo "copy them from the database sources into /usr/local/bin"
-         exit 1
-      else
-         echo "yes"
-      fi
-   else
-      echo "yes"
-   fi
-fi
 
 # APQ will not build with no database selected.  If no database is selected,
 # skip this step.  Otherwise, select one database or both through the APQ
@@ -1230,14 +1211,14 @@
 # NO_MYSQL means the user requests no MSYQL support, not that MySQL is not
 # installed (that is HAS_MYSQL).
 
-if [ -n "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
+if [ -z "$HAS_MYSQL" -a -z "$HAS_POSTGRES" ] ; then
     echo "No MySQL or PostgreSQL support."
 else
    cd apq-2.1
-   if [ -n "$NO_MYSQL" -a -z "$NO_POSTGRES" ] ; then
+   if [ -z "$HAS_MYSQL" -a -n "$HAS_POSTGRES" ] ; then
       echo "Running APQ's configure (no MySQL)..."
       HAVE_MY=0 ./configure
-   elif [ -z "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
+   elif [ -n "$HAS_MYSQL" -a -z "$HAS_POSTGRES" ] ; then
       echo "Running APQ's configure (no PostgreSQL)..."
       HAVE_PG=0 ./configure
    else
@@ -1480,7 +1461,7 @@
    RLINCL="\-I\.\/areadline\/"
    # Linux Mint doesn't need ncurses and breaks...needed for Slackware?
    #RLLIBS="\-L\/lib\/x86_64\-linux\-gnu\/\ \-l:libreadline.so.6\ \-l:libhistory.so.6\ \-lncurses"
-   RLLIBS="\-L\/lib\/x86_64\-linux\-gnu\/\ \-l:libreadline.so.$READLINE_VERSION\ \-l:libhistory.so.$READLINE_VERSION"
+   RLLIBS="\-lreadline\ \-lhistory\ \-lncurses"
 else
    RLINCL=
    RLLIBS=
diff -ru SparForte-2.1/src/apq-2.1/Makefile SparForte-2.1-new/src/apq-2.1/Makefile
--- SparForte-2.1/src/apq-2.1/Makefile	2017-09-01 17:51:22.000000000 -0700
+++ SparForte-2.1-new/src/apq-2.1/Makefile	2017-09-19 13:29:18.979754152 -0700
@@ -40,7 +40,7 @@
 	ADA_INCLUDE_PATH=. ADA_OBJECTS_PATH=. gnatmake -c $(AOPTS) comp_mysql
 
 c_mysql.o: c_mysql.c
-	$(CC) -c $(COPTS) $(ZOPTS) $(MY_INCL) c_mysql.c -o c_mysql.o
+	$(CC) -c $(COPTS) $(ZOPTS) $(MY_INCL) -I/usr/include/mysql/server c_mysql.c -o c_mysql.o
 
 apq-mysql.ads: apq-mysql.ads-in mysql_generr.c mysql_xcr2 mysql_xcr3
 	@rm -f mysql_errmsg.h mysql_generr
