--- SparForte-2.2/configure	2019-01-04 14:31:18.000000000 -0800
+++ SparForte-2.2-new/configure	2019-01-26 06:49:44.564677232 -0800
@@ -1086,66 +1086,8 @@
 HAS_READLINE=
 READLINE_VERSION=
 if [ -z "$NO_READLINE" ] ; then
-   echo -n "checking for readline 6 (ldconfig -p, Linux)... "
-   TMP=`$LDCONFIG_SUDO /sbin/ldconfig -p 2>/dev/null | grep libreadline.so.6`
-   if [ -z "$TMP" ] ; then
-      echo "no"
-   else
-      HAS_READLINE=1
-      READLINE_VERSION=6
-      echo "yes"
-   fi
-   if [ -z "$READLINE_VERSION" ] ; then
-      echo -n "checking for readline 7 (ldconfig -p, Linux)... "
-      TMP=`$LDCONFIG_SUDO sudo /sbin/ldconfig -p 2>/dev/null | grep libreadline.so.7`
-      if [ -z "$TMP" ] ; then
-         echo "no"
-      else
-         HAS_READLINE=1
-         READLINE_VERSION=7
-         echo "yes"
-      fi
-   fi
-   if [ -z "$READLINE_VERSION" ] ; then
-      echo -n "checking for readline (ldconfig -r, FreeBSD)... "
-      TMP=`/sbin/ldconfig -r 2>/dev/null | grep libreadline`
-      if [ -z "$TMP" ] ; then
-         echo "no"
-      else
-         HAS_READLINE=1
-         READLINE_VERSION=6
-         echo "yes"
-      fi
-   fi
-   if [ -z "$READLINE_VERSION" ] ; then
-      if [ "$OS" = "cygwin" ] ; then
-         echo -n "checking for readline (locate, Cygwin)... "
-         TMP=`"$LOCATE" libreadline`
-         if [ -z "$TMP" ] ; then
-            echo "no"
-         else
-            HAS_READLINE=1
-            READLINE_VERSION=
-            echo "yes"
-         fi
-      else
-         echo -n "checking for readline (locate, Darwin)... "
-         TMP=`"$LOCATE" libreadline`
-         if [ -z "$TMP" ] ; then
-            echo "no"
-         else
-            HAS_READLINE=1
-            READLINE_VERSION=6
-            echo "yes"
-         fi
-      fi
-   fi
-   if [ -z "$HAS_READLINE" ] ; then
-      echo
-      echo "SparForte cannot find GNU readline and history libraries"
-      echo "Please install them or run configure with --without-readline"
-      exit 1
-   fi
+    HAS_READLINE=1
+    READLINE_VERSION=7
 fi
 
 # GNU Make
@@ -1252,25 +1194,6 @@
 sleep 1
 
 echo -n "apq library: checking for one or more database config scripts... "
-if [ -n "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
-   echo "no"
-else
-   if [ -z "$HAS_POSTGRES" ] ; then
-      if [ -z "$HAS_MYSQL" ] ; then
-         echo "no"
-         echo
-         echo "The APQ library uses the pg_config and mysql_config scripts to configure"
-         echo "itself.  Not all operating systems include these, or they include them"
-         echo "as part of the 'devel' packages.  If you don't have these scripts, you"
-         echo "copy them from the database sources into /usr/local/bin"
-         exit 1
-      else
-         echo "yes"
-      fi
-   else
-      echo "yes"
-   fi
-fi
 
 # APQ will not build with no database selected.  If no database is selected,
 # skip this step.  Otherwise, select one database or both through the APQ
@@ -1280,14 +1203,14 @@
 # NO_MYSQL means the user requests no MSYQL support, not that MySQL is not
 # installed (that is HAS_MYSQL).
 
-if [ -n "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
+if [ -z "$HAS_MYSQL" -a -z "$HAS_POSTGRES" ] ; then
     echo "No MySQL or PostgreSQL support."
 else
    cd apq-2.1
-   if [ -n "$NO_MYSQL" -a -z "$NO_POSTGRES" ] ; then
+   if [ -z "$HAS_MYSQL" -a -n "$HAS_POSTGRES" ] ; then
       echo "Running APQ's configure (no MySQL)..."
       HAVE_MY=0 ./configure
-   elif [ -z "$NO_MYSQL" -a -n "$NO_POSTGRES" ] ; then
+   elif [ -n "$HAS_MYSQL" -a -z "$HAS_POSTGRES" ] ; then
       echo "Running APQ's configure (no PostgreSQL)..."
       HAVE_PG=0 ./configure
    else
@@ -1535,7 +1458,7 @@
    if [ -z "$READLINE_VERSION" ] ; then
       RLLIBS="\-L\/lib\/x86_64\-linux\-gnu\/\ \-lreadline\ \-lhistory"
    else
-      RLLIBS="\-L\/lib\/x86_64\-linux\-gnu\/\ \-l:libreadline.so.$READLINE_VERSION\ \-l:libhistory.so.$READLINE_VERSION"
+      RLLIBS="\-L\/lib\/x86_64\-linux\-gnu\/\ \-lreadline\ \-lhistory \-ltinfo"
    fi
 else
    RLINCL=
