diff --git a/src/apq-2.1/mysql_xty b/src/apq-2.1/mysql_xty
index f9c420c..ed7c34d 100755
--- a/src/apq-2.1/mysql_xty
+++ b/src/apq-2.1/mysql_xty
@@ -4,7 +4,7 @@ PATH=".:$PATH"
 
 HDRFILE=$(mysql_incl)/mysql_com.h
 
-sed <$HDRFILE -n '/enum_field_types/,/};/p' | sed 's|enum||;s|enum_field_types||;s|[{};]||g;s|,|\
+cpp -P $HDRFILE | sed -n '/enum_field_types/,/};/p' | sed 's|enum||;s|enum_field_types||;s|[{};]||g;s|,|\
 |g;s|[ 	]*||g' | sed '/^$/d;s|=[0-9]*||g' \
         | while read NAME ; do
                 echo "  { \"$NAME\", $NAME },"
