--- gps-6.1.1-src/Makefile.in	2014-11-13 10:15:57.000000000 -0800
+++ gps-6.1.1-new/Makefile.in	2015-09-04 17:54:03.244791168 -0700
@@ -10,9 +10,9 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA    = @INSTALL_DATA@
 prefix          = @prefix@
-bindir          = $(prefix)/bin
-sharedir        = $(prefix)/share
-examplesdir     = $(prefix)/share/examples/gps
+bindir          = $(DESTDIR)$(prefix)/bin
+sharedir        = $(DESTDIR)$(prefix)/share
+examplesdir     = $(DESTDIR)$(prefix)/share/examples/gps
 
 all default:
 	$(MAKE) -C gps default
