diff -ur gps-git261227eb71714.org/common/common_no_xmlada.gpr.in gps-git261227eb71714/common/common_no_xmlada.gpr.in
--- gps-git261227eb71714.org/common/common_no_xmlada.gpr.in	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/common/common_no_xmlada.gpr.in	2012-04-14 09:08:06.498043208 -0700
@@ -1,5 +1,4 @@
 with "gtkada";
-with "../gnat";
 with "../shared";
 with "../gnatlib/src/gnatcoll_gtk";
 
diff -ur gps-git261227eb71714.org/common/common_with_xmlada.gpr.in gps-git261227eb71714/common/common_with_xmlada.gpr.in
--- gps-git261227eb71714.org/common/common_with_xmlada.gpr.in	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/common/common_with_xmlada.gpr.in	2012-04-14 09:07:33.985055207 -0700
@@ -1,5 +1,4 @@
 with "gtkada";
-with "../gnat";
 with "../shared";
 with "xmlada_sax";
 with "../gnatlib/src/gnatcoll_gtk";
diff -ur gps-git261227eb71714.org/gnatlib/Makefile gps-git261227eb71714/gnatlib/Makefile
--- gps-git261227eb71714.org/gnatlib/Makefile	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gnatlib/Makefile	2012-04-14 08:16:50.685935319 -0700
@@ -10,7 +10,7 @@
 install: install_static install_docs
 endif
 
-include Makefile.gnat
+#include Makefile.gnat
 
 ## Builds explicitly the shared or the static libraries
 
@@ -65,7 +65,7 @@
 	-@$(foreach f,$(GNAT_SOURCES_FOR_GNATCOLL), \
 	   $(LN_S) ../gnat_src/$(f) gnat >/dev/null 2>&1 ;)
 endif
-	@(cd gnat && gnatmake -q xsnamest && ./xsnamest && mv snames.ns snames.ads && mv snames.nb snames.adb)
+#	@(cd gnat && gnatmake -q xsnamest && ./xsnamest && mv snames.ns snames.ads && mv snames.nb snames.adb)
 
 else
 do_links:
diff -ur gps-git261227eb71714.org/gnatlib/gnatcoll_build.gpr gps-git261227eb71714/gnatlib/gnatcoll_build.gpr
--- gps-git261227eb71714.org/gnatlib/gnatcoll_build.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gnatlib/gnatcoll_build.gpr	2012-04-14 08:18:27.301938800 -0700
@@ -1,5 +1,7 @@
 --  This project file is used to build all modules that have no external
 --  dependencies on external project files
+with "gnatprj";
+with "gnatvsn";
 with "src/gnatcoll";
 with "src/gnatcoll_python";
 with "src/gnatcoll_postgres";
@@ -8,6 +10,7 @@
 with "src/gnatcoll_gmp";
 with "src/gnatcoll_readline";
 with "gnatcoll_shared";
+
 project Gnatcoll_Build is
    for Source_Dirs use ();
    for Object_Dir use "src/" & Gnatcoll_Shared'Object_Dir;  --  for auto.cgpr
diff -ur gps-git261227eb71714.org/gnatlib/gnatcoll_shared.gpr.in gps-git261227eb71714/gnatlib/gnatcoll_shared.gpr.in
--- gps-git261227eb71714.org/gnatlib/gnatcoll_shared.gpr.in	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gnatlib/gnatcoll_shared.gpr.in	2012-04-14 08:21:32.730944969 -0700
@@ -1,3 +1,5 @@
+with "gnatvsn";
+with "gnatprj";
 project GnatColl_Shared is
    Version := "@PACKAGE_VERSION@";
    Name    := "@PACKAGE_TARNAME@";
diff -ur gps-git261227eb71714.org/gnatlib/src/Makefile.gnatcoll gps-git261227eb71714/gnatlib/src/Makefile.gnatcoll
--- gps-git261227eb71714.org/gnatlib/src/Makefile.gnatcoll	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gnatlib/src/Makefile.gnatcoll	2012-04-14 07:55:28.519891453 -0700
@@ -10,6 +10,6 @@
 SRCDIRS=./ ./nosyslog/ ./paragraph_filling/
 endif
 
-ifeq ($(GNAT_SOURCES),copy)
-SRCDIRS+= ../gnat/
-endif
+#ifeq ($(GNAT_SOURCES),copy)
+#SRCDIRS+= ../gnat/
+#endif
diff -ur gps-git261227eb71714.org/gps/Makefile gps-git261227eb71714/gps/Makefile
--- gps-git261227eb71714.org/gps/Makefile	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gps/Makefile	2012-04-14 08:39:34.953981994 -0700
@@ -19,7 +19,7 @@
 
 default: resources do_links all
 
-include ../Makefile.gnat
+#include ../Makefile.gnat
 
 # NOTE: we need to build gnatcoll separately, since we cannot use
 # gnat.adc (No_Tasking restriction) on all gnatcoll sources.
diff -ur gps-git261227eb71714.org/gps/gps.gpr gps-git261227eb71714/gps/gps.gpr
--- gps-git261227eb71714.org/gps/gps.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gps/gps.gpr	2012-04-14 08:40:13.319983308 -0700
@@ -36,6 +36,7 @@
 with "../views/views";
 with "../vfs/vfs";
 with "../vsearch/vsearch";
+with "gnatprj";
 
 project GPS is
 
diff -ur gps-git261227eb71714.org/kernel/kernel.gpr gps-git261227eb71714/kernel/kernel.gpr
--- gps-git261227eb71714.org/kernel/kernel.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/kernel/kernel.gpr	2012-04-14 09:09:18.509044853 -0700
@@ -1,5 +1,4 @@
 --  The part of the kernel that depends on the GUI
-with "../gnat";
 with "gtkada";
 with "kernel_core";
 with "shared";
diff -ur gps-git261227eb71714.org/kernel/kernel_core.gpr gps-git261227eb71714/kernel/kernel_core.gpr
--- gps-git261227eb71714.org/kernel/kernel_core.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/kernel/kernel_core.gpr	2012-04-14 09:10:19.453044402 -0700
@@ -1,4 +1,3 @@
-with "../gnat";
 with "../common/common";
 with "../shared";
 with "../gnatlib/src/gnatcoll";
diff -ur gps-git261227eb71714.org/shared.gpr gps-git261227eb71714/shared.gpr
--- gps-git261227eb71714.org/shared.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/shared.gpr	2012-04-14 09:15:25.475055632 -0700
@@ -35,7 +35,7 @@
                "-gnatwaCJe", "-gnat05", "-gnateE");
 
             for Default_Switches ("C") use
-              ("-g", "-O1", "-fdump-xref") & GnatColl_Shared.Gtk_Include;
+              ("-g", "-O1") & GnatColl_Shared.Gtk_Include;
 
          when "Production" =>
             for Default_Switches ("Ada") use
--- gps-git261227eb71714.org/gnatlib/src/gnatcoll__copy.gpr	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/gnatlib/src/gnatcoll__copy.gpr	2012-04-14 12:02:17.003398087 -0700
@@ -12,9 +12,7 @@
       "lib" & Project'Library_Name & ".so." & GnatColl_Shared.Version;
 
    case Gnatcoll_Shared.Projects is
-      when "yes" =>
-         for Source_Dirs use
-            Project'Source_Dirs & ("../gnat", "../gnat/sdefault");
+      when "yes" => null;
       when "no" =>
          for Excluded_Source_Files use ("gnatcoll-projects.ads",
                                         "gnatcoll-projects.adb",
@@ -56,8 +54,7 @@
       end case;
 
       case GnatColl_Shared.Projects is
-         when "yes" => for Implementation ("mlib.tgt.specific")
-            use "mlib-tgt-specific-linux.adb";
+         when "yes" =>  null;
          when "no" =>
             null;
       end case;
--- gps-git261227eb71714.org/distrib/from_impunit.pl	1969-12-31 16:00:00.000000000 -0800
+++ gps-git261227eb71714/distrib/from_impunit.pl	2012-04-14 16:49:42.845988110 -0700
@@ -7,7 +7,7 @@
 $expand_hierarchies = 1;
 
 $output ="../share/plug-ins/runtime.xml";
-$impunit="../gnatlib/gnat_src/impunit.adb";
+$impunit="/usr/include/gnatprj/impunit.adb";
 
 open (OUT, '>' . $output);
 open (IMPUNIT, $impunit) || die "File $impunit not found";
